cloudferrylib/os/actions/check_networks.py

```diff
@@ -88,8 +85,226 @@ def run(self, **kwargs):
             LOG.critical('Network overlapping list:\n%s', overlapped_resources)
             raise exception.AbortMigrationError(
                 "There is a number of overlapping Network resources, so "
-                "migration process can not be continued. Resolve it please and"
+                "migration process can not "
+                "be continued. Resolve it please and"
                 " try again.")
+        self.check_conflicts_ports(src_net,
+                                   dst_net,
+                                   kwargs.get('search_opts', {}))
+        
+    def get_similar_tenants(self):
+        src_identity = self.src_cloud.resources[utils.IDENTITY_RESOURCE]
+        dst_identity = self.dst_cloud.resources[utils.IDENTITY_RESOURCE]
+        src_tenants = src_identity.read_info()['tenants']
+        dst_tenants = {t['tenant']['name']: t['tenant']['id']
+                       for t in dst_identity.read_info()['tenants']}
+        similar_tenants = {}
+        for ts in src_tenants:
+            index = ts['tenant']['name']
+            if index in dst_tenants:
+                similar_tenants[ts['tenant']['id']] = dst_tenants[index]
+            else:
+                similar_tenants[ts['tenant']['id']] = ''
+        return similar_tenants
+
+    def get_instances(self, cloud, search_opts={}):
+        compute = cloud.resources[utils.COMPUTE_RESOURCE]
+        instances = compute.read_info('instances',
+                                      search_opts=
+                                      search_opts)['instances']
+        return instances
+
+    def get_similar_instances(self, instances):
```

Comment:
I'm not sure how tenants and instances became part of `NetworkCheck` object, seem separate object-wise, I guess it's a code smell